### PR TITLE
Better backtraces

### PR DESCRIPTION
**Pull request by brendanlong**
See https://caml.inria.fr/mantis/print_bug_page.php?bug_id=7375

### CODE REVIEW

**Comment by brendanlong** [comment]
Actually 4.05 works fine in our codebase so I'm just going to bump the version here.

**Comment by brendanlong** [comment]
For reference, this fixes backtraces that used to look like:

```
Ingest failed: (monitor.ml.Error (Failure "Month over month metrics regression")
 ("Raised at file \"pgx/pgx/src/pgx.ml\", line 1399, characters 9-50"
  "Called from file \"src/deferred0.ml\", line 61, characters 64-69"
  "Called from file \"src/job_queue.ml\", line 159, characters 6-47"))
```

And now look like:

```
Ingest failed: (monitor.ml.Error (Failure "Month over month metrics regression")
 ("Raised at file \"pervasives.ml\", line 32, characters 17-33"
  "Called from file \"genisys/src/ingest_validation.ml\", line 168, characters 6-58"
  "Called from file \"src/deferred0.ml\", line 61, characters 64-69"
  "Called from file \"src/job_queue.ml\", line 159, characters 6-47"
  "Re-raised at file \"pgx/pgx/src/pgx.ml\", line 1399, characters 9-50"
  "Called from file \"src/deferred0.ml\", line 61, characters 64-69"
  "Called from file \"src/job_queue.ml\", line 159, characters 6-47"))
```

File: pgx_unix/src/pgx_unix.ml [hunk 1]
```diff
@@ -19,11 +19,12 @@
  * Boston, MA 02111-1307, USA.
 *)
 
+external reraise : exn -> _ = "%reraise"
+
 module Simple_thread = struct
   type 'a t = 'a
   let return x = x
   let (>>=) v f =  f v
-  let fail = raise
```
Comment (kevinqiu) [inline pgx_unix/src/pgx_unix.ml]: Why did we do this alias before?

Comment (brendanlong) [reply pgx_unix/src/pgx_unix.ml]: No idea.. this was before my time. I just noticed that it's completely useless now :\

**Pull request merged**